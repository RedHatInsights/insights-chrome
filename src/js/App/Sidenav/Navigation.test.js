import React from 'react';
import ConnectedNavigation, { Navigation, dispatchToProps }  from './Navigation';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

let settingsData = [{ id: 'overview', title: 'Overview', group: 'insights', active: true, subItems:
    [{ id: 'someID', title: 'Clusters', default: true }] },
{ id: 'rules', title: 'Rules', group: 'insights', active: false },
{ id: 'topics', title: 'Topics', group: 'insights', active: false },
{ title: 'Inventory', id: 'inventory', active: false },
{ title: 'Remediations', id: 'remediations', active: false }];

describe('Navigation', () => {
    const initialProps = {
        settings: settingsData,
        activeApp: 'someApp',
        activeLocation: 'openshift',
        documentation: 'someDocs'
    };
    let initialState;
    let mockStore;
    let globalNavData = require('../../../../testdata/globalNav.json');

    beforeEach(() =>{
        mockStore = configureStore();
        initialState = ({
            chrome: {
                globalNav: [globalNavData],
                activeApp: 'someApp',
                navHidden: false,
                activeLocation: 'someLocation',
                activeGroup: 'someGroup',
                appId: 'someId'
            }
        });
        const pathname = '/';
        Object.defineProperty(window, 'location', {
            value: {
                pathname: pathname
            },
            writable: true
        });
    });
    it('should render corectly', () =>{
        let props = {
            ...initialProps,
            appId: 'overview',
            activeGroup: 'someID'
        };
        const mockSelect = jest.fn();
        const mockClick = jest.fn();
        const store = mockStore(initialState);
        const wrapper = shallow(<Navigation onSelect={ mockSelect } onClick={ mockClick } store={ store }{ ...props }/>);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.find(`[itemID='someID']`).simulate('click', { persist: jest.fn() });
        wrapper.find(`[itemID='rules']`).simulate('click', { persist: jest.fn() });
        wrapper.find(`[aria-label='Insights Global Navigation']`).simulate('select', { groupId: 'someID1', itemID: 'someID2' });

    });
    it('should render correctly 2', () =>{
        let props = {
            ...initialProps,
            appId: 'overview',
            activeGroup: 'insights'
        };
        const mockSelect = jest.fn();
        const mockClick = jest.fn();
        const mockNavigate = jest.fn();
        const mockClear = jest.fn();
        const store = mockStore(initialState);
        const wrapper = shallow(<Navigation onSelect= { mockSelect } onClick={ mockClick }
            onNavigate={ mockNavigate } onClearActive = { mockClear } store={ store }{ ...props }/>);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.find(`[itemID='rules']`).simulate('click', { persist: jest.fn() });

    });
});

describe('ConnectedNavigation', () => {
    let initialState;
    let mockStore;
    let globalNavData = require('../../../../testdata/globalNav.json');

    beforeEach(() =>{
        mockStore = configureStore();
        initialState = ({
            chrome: {
                globalNav: [globalNavData],
                activeApp: 'someApp',
                navHidden: false,
                activeLocation: 'someLocation',
                activeGroup: 'someGroup',
                appId: 'someId'
            }
        });
    });

    it('should render correctly with initial state', () =>{
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <ConnectedNavigation/>
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('mapDispatchToProps function fires', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <ConnectedNavigation store= {store}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
        const mockDispatch = jest.fn();
        const actionProps = dispatchToProps(mockDispatch);
        actionProps.onNavigate(jest.fn(), jest.fn());
        actionProps.onClearActive();
        expect(mockDispatch.mock.calls.length).toBe(2);
    });

});
