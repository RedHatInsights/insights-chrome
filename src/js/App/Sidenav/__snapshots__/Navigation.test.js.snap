// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedNavigation mapDispatchToProps function fires 1`] = `
<Connect(Navigation)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Navigation
    activeApp="someApp"
    activeGroup="someGroup"
    activeLocation="someLocation"
    appId="someId"
    navHidden={false}
    onClearActive={[Function]}
    onNavigate={[Function]}
    settings={
      Array [
        Object {
          "ansible": Object {
            "id": "ansible",
            "routes": Array [],
            "title": "Red Hat Ansible Automation Platform",
          },
          "docs": Object {
            "id": "docs",
            "routes": Array [],
            "title": "Documentation",
          },
          "insights": Object {
            "id": "insights",
            "routes": Array [],
            "title": "Red Hat Insights",
          },
          "openshift": Object {
            "id": "openshift",
            "routes": Array [],
            "title": "Red Hat Openshift Cluster Manager",
          },
          "rhel": Object {
            "id": "rhel",
            "routes": Array [],
            "title": "Cloud Management Services",
          },
          "staging": Object {
            "id": "staging",
            "routes": Array [],
            "title": "Staging Bundle",
          },
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Component
      aria-label="Insights Global Navigation"
      data-ouia-safe="true"
      onSelect={[Function]}
      theme="dark"
    >
      <ComponentWithOuia
        component={[Function]}
        componentProps={
          Object {
            "aria-label": "Insights Global Navigation",
            "children": <NavList
              ariaLeftScroll="Scroll left"
              ariaRightScroll="Scroll right"
              className=""
              variant="default"
            >
              <NavigationItem
                isActive={false}
                onClick={[Function]}
                parent="someLocation"
              />
            </NavList>,
            "data-ouia-safe": "true",
            "onSelect": [Function],
            "theme": "dark",
          }
        }
        consumerContext={null}
      >
        <Nav
          aria-label="Insights Global Navigation"
          className=""
          data-ouia-safe="true"
          onSelect={[Function]}
          onToggle={[Function]}
          ouiaContext={
            Object {
              "isOuia": false,
              "ouiaId": null,
            }
          }
          theme="dark"
        >
          <nav
            aria-label="Insights Global Navigation"
            className="pf-c-nav pf-m-dark"
            data-ouia-safe="true"
          >
            <NavList
              ariaLeftScroll="Scroll left"
              ariaRightScroll="Scroll right"
              className=""
              variant="default"
            >
              <ul
                className="pf-c-nav__list"
              >
                <NavigationItem
                  isActive={false}
                  key="0"
                  onClick={[Function]}
                  parent="someLocation"
                >
                  <NavItem
                    isActive={false}
                    onClick={[Function]}
                    preventDefault={true}
                    to="http://localhost/someLocation/undefined"
                  >
                    <li
                      className="pf-c-nav__item"
                    >
                      <a
                        aria-current={null}
                        className="pf-c-nav__link"
                        href="http://localhost/someLocation/undefined"
                        onClick={[Function]}
                      />
                    </li>
                  </NavItem>
                </NavigationItem>
              </ul>
            </NavList>
          </nav>
        </Nav>
      </ComponentWithOuia>
    </Component>
  </Navigation>
</Connect(Navigation)>
`;

exports[`ConnectedNavigation should render correctly with initial state 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Navigation)>
    <Navigation
      activeApp="someApp"
      activeGroup="someGroup"
      activeLocation="someLocation"
      appId="someId"
      navHidden={false}
      onClearActive={[Function]}
      onNavigate={[Function]}
      settings={
        Array [
          Object {
            "ansible": Object {
              "id": "ansible",
              "routes": Array [],
              "title": "Red Hat Ansible Automation Platform",
            },
            "docs": Object {
              "id": "docs",
              "routes": Array [],
              "title": "Documentation",
            },
            "insights": Object {
              "id": "insights",
              "routes": Array [],
              "title": "Red Hat Insights",
            },
            "openshift": Object {
              "id": "openshift",
              "routes": Array [],
              "title": "Red Hat Openshift Cluster Manager",
            },
            "rhel": Object {
              "id": "rhel",
              "routes": Array [],
              "title": "Cloud Management Services",
            },
            "staging": Object {
              "id": "staging",
              "routes": Array [],
              "title": "Staging Bundle",
            },
          },
        ]
      }
    >
      <Component
        aria-label="Insights Global Navigation"
        data-ouia-safe="true"
        onSelect={[Function]}
        theme="dark"
      >
        <ComponentWithOuia
          component={[Function]}
          componentProps={
            Object {
              "aria-label": "Insights Global Navigation",
              "children": <NavList
                ariaLeftScroll="Scroll left"
                ariaRightScroll="Scroll right"
                className=""
                variant="default"
              >
                <NavigationItem
                  isActive={false}
                  onClick={[Function]}
                  parent="someLocation"
                />
              </NavList>,
              "data-ouia-safe": "true",
              "onSelect": [Function],
              "theme": "dark",
            }
          }
          consumerContext={null}
        >
          <Nav
            aria-label="Insights Global Navigation"
            className=""
            data-ouia-safe="true"
            onSelect={[Function]}
            onToggle={[Function]}
            ouiaContext={
              Object {
                "isOuia": false,
                "ouiaId": null,
              }
            }
            theme="dark"
          >
            <nav
              aria-label="Insights Global Navigation"
              className="pf-c-nav pf-m-dark"
              data-ouia-safe="true"
            >
              <NavList
                ariaLeftScroll="Scroll left"
                ariaRightScroll="Scroll right"
                className=""
                variant="default"
              >
                <ul
                  className="pf-c-nav__list"
                >
                  <NavigationItem
                    isActive={false}
                    key="0"
                    onClick={[Function]}
                    parent="someLocation"
                  >
                    <NavItem
                      isActive={false}
                      onClick={[Function]}
                      preventDefault={true}
                      to="http://localhost/someLocation/undefined"
                    >
                      <li
                        className="pf-c-nav__item"
                      >
                        <a
                          aria-current={null}
                          className="pf-c-nav__link"
                          href="http://localhost/someLocation/undefined"
                          onClick={[Function]}
                        />
                      </li>
                    </NavItem>
                  </NavigationItem>
                </ul>
              </NavList>
            </nav>
          </Nav>
        </ComponentWithOuia>
      </Component>
    </Navigation>
  </Connect(Navigation)>
</Provider>
`;

exports[`Navigation should render corectly 1`] = `
<Component
  aria-label="Insights Global Navigation"
  data-ouia-safe="true"
  onSelect={[Function]}
  theme="dark"
>
  <NavList
    ariaLeftScroll="Scroll left"
    ariaRightScroll="Scroll right"
    className=""
    variant="default"
  >
    <NavExpandable
      className=""
      groupId={null}
      id=""
      isActive={true}
      isExpanded={true}
      itemID="overview"
      key="0"
      ouia-nav-group="overview"
      srText=""
      title="Overview"
    >
      <NavigationItem
        isActive={false}
        itemID="someID"
        key="0"
        onClick={[Function]}
        ouia-nav-item="someID"
        parent="openshift/overview"
        title="Clusters"
      />
    </NavExpandable>
    <NavigationItem
      isActive={false}
      itemID="rules"
      key="1"
      onClick={[Function]}
      ouia-nav-item="rules"
      parent="openshift"
      title="Rules"
    />
    <NavigationItem
      isActive={false}
      itemID="topics"
      key="2"
      onClick={[Function]}
      ouia-nav-item="topics"
      parent="openshift"
      title="Topics"
    />
    <NavigationItem
      isActive={false}
      itemID="inventory"
      key="3"
      onClick={[Function]}
      ouia-nav-item="inventory"
      parent="openshift"
      title="Inventory"
    />
    <NavigationItem
      isActive={false}
      itemID="remediations"
      key="4"
      onClick={[Function]}
      ouia-nav-item="remediations"
      parent="openshift"
      title="Remediations"
    />
    <NavItemSeparator />
    <NavItem
      className="ins-c-page__documentation"
      rel="noopener noreferrer"
      target="_blank"
      to="someDocs"
    >
      Documentation
    </NavItem>
    <NavItem
      key="supportcases"
      rel="noopener noreferrer"
      target="_blank"
      to="https://access.redhat.com/support/cases"
    >
      Support Cases
    </NavItem>
    <NavItem
      key="feedback"
      rel="noopener noreferrer"
      target="_blank"
      to="mailto:ocm-feedback@redhat.com"
    >
      Cluster Manager Feedback
    </NavItem>
    <NavItem
      key="bugs"
      rel="noopener noreferrer"
      target="_blank"
      to="https://bugzilla.redhat.com/enter_bug.cgi?product=OpenShift%20Container%20Platform"
    >
      Report an OpenShift Bug
    </NavItem>
  </NavList>
</Component>
`;

exports[`Navigation should render correctly 2 1`] = `
<Component
  aria-label="Insights Global Navigation"
  data-ouia-safe="true"
  onSelect={[Function]}
  theme="dark"
>
  <NavList
    ariaLeftScroll="Scroll left"
    ariaRightScroll="Scroll right"
    className=""
    variant="default"
  >
    <NavExpandable
      className=""
      groupId={null}
      id=""
      isActive={true}
      isExpanded={true}
      itemID="overview"
      key="0"
      ouia-nav-group="overview"
      srText=""
      title="Overview"
    >
      <NavigationItem
        isActive={false}
        itemID="someID"
        key="0"
        onClick={[Function]}
        ouia-nav-item="someID"
        parent="openshift/overview"
        title="Clusters"
      />
    </NavExpandable>
    <NavigationItem
      isActive={false}
      itemID="rules"
      key="1"
      onClick={[Function]}
      ouia-nav-item="rules"
      parent="openshift"
      title="Rules"
    />
    <NavigationItem
      isActive={false}
      itemID="topics"
      key="2"
      onClick={[Function]}
      ouia-nav-item="topics"
      parent="openshift"
      title="Topics"
    />
    <NavigationItem
      isActive={false}
      itemID="inventory"
      key="3"
      onClick={[Function]}
      ouia-nav-item="inventory"
      parent="openshift"
      title="Inventory"
    />
    <NavigationItem
      isActive={false}
      itemID="remediations"
      key="4"
      onClick={[Function]}
      ouia-nav-item="remediations"
      parent="openshift"
      title="Remediations"
    />
    <NavItemSeparator />
    <NavItem
      className="ins-c-page__documentation"
      rel="noopener noreferrer"
      target="_blank"
      to="someDocs"
    >
      Documentation
    </NavItem>
    <NavItem
      key="supportcases"
      rel="noopener noreferrer"
      target="_blank"
      to="https://access.redhat.com/support/cases"
    >
      Support Cases
    </NavItem>
    <NavItem
      key="feedback"
      rel="noopener noreferrer"
      target="_blank"
      to="mailto:ocm-feedback@redhat.com"
    >
      Cluster Manager Feedback
    </NavItem>
    <NavItem
      key="bugs"
      rel="noopener noreferrer"
      target="_blank"
      to="https://bugzilla.redhat.com/enter_bug.cgi?product=OpenShift%20Container%20Platform"
    >
      Report an OpenShift Bug
    </NavItem>
  </NavList>
</Component>
`;
